nipype/pipeline/plugins/somaflow.py

```diff
@@ -27,9 +27,8 @@ def _submit_graph(self, pyfiles, dependencies, nodes):
         for fname in pyfiles:
             name = os.path.splitext(os.path.split(fname)[1])[0]
             jobs.append(Job(command=[sys.executable, fname], name=name))
-        for key, values in list(dependencies.items()):
-            for val in values:
-                soma_deps.append((jobs[val], jobs[key]))
+        for key, values in dependencies.items():
+            soma_deps.extend((jobs[val], jobs[key]) for val in values)
```

Comment:
I think this one can be turned into one extend with a doubly nested for loop.

Reply:
Indeed, I modified this piece of code accordingly